DSEGOG-52 multiple channels in plot

sam-glendenning opened the pull request:
## Description
This allows for multiple data channels to be plotted in the y-axis. I've removed the <Select> box for choosing channels as it basically does the same thing as the <Autocomplete> but with less functionality so it just gets in the way. This should allow for slightly easier channel selection.

Due to the wide variety in scalar values, plots can look a little weird if totally unrelated channels are plotted side by side. For best results, use two related ones, such as {something}-XPOS and {something}-YPOS.

## Testing instructions
Try plotting multiple channels in the y-axis. Scatter and line graphs should both work and produce nice looking plots.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes DSEGOG-52

Commit:
DSEGOG-52 multiple channel labels in settings

This allows the user to add multiple channels to plot in the plot settings on the y-axis. All this does right now is display a label for each selected channel, this isn't passed to the plot yet. I've also removed the <Select> for choosing a channel, as I feel it gets in the way and the <Autocomplete> does a better job of it anyway

Commit:
Merge branch 'develop' into DSEGOG-52-multiple-channels-in-plot

Commit:
DSEGOG-52 multiple channels can now be plotted

We pass an array of datasets to the <Plot> component and map each one to a VictoryLine and VictoryScatter where sufficient, plus a legend label for each one

Commit:
Merge branch 'develop' into DSEGOG-52-multiple-channels-in-plot

Commit:
Merge branch 'develop' into DSEGOG-52-multiple-channels-in-plot

Commit:
DSEGOG-52 fixing linting

Commit:
DSEGOG-52 updating plot settings tests

Commit:
Merge branch 'develop' into DSEGOG-52-multiple-channels-in-plot

Commit:
Updating records.tsx to use selectedChannels and not generic YAxis variable

Commit:
DSEGOG-52 enhancing plot data exporting to allow for multiple channels

Multi-channel plots can now be exported properly as CSV files

Commit:
DSEGOG-52 removing milliseconds from timestamps and adding comments

Commit:
DSEGOG-52 minor test fixes

Commit:
DESGOG-52 further fixes and adding testPlotRecords

This can be used in other plotting tests later

Commit:
DSEGOG-52 timestamp no longer plottable as data channel

Can no longer have time appear in y-axis. This correlates with how CLF want to plot a time-series graph

Commit:
DSEGOG-52 remove old code

Commit:
DSEGOG-52 fixing resize plot test

Commit:
DSEGOG-52 bumping coverage for plot buttons

sam-glendenning commented on src/plotting/plot.component.tsx:
```diff
@@ -8,39 +8,39 @@ import {
+  XAxisSettings,
+  PlotDataset,
+  PlotType,
+  YAxisSettings,
+} from '../app.types';
 import { format } from 'date-fns';
 
-export const formatTooltipLabel = (label: number | Date): number | string => {
-  if (label instanceof Date) {
+export const formatTooltipLabel = (
+  label: number,
+  scale: XAxisSettings['scale']
+): number | string => {
+  if (scale === 'time') {
     return format(label, 'yyyy-MM-dd HH:mm:ss');
   }
   return label;
 };
```
I've gone back to just using numbers only for all labels, including timestamps, hence why you no longer see 
```
{ [channel: string]: number | Date }
```
This just makes it much easier to manage the sorting and ordering of data. This function is called when viewing a VictoryLabel and when converting unix timestamps for export to CSV, which are the only instances unix timestamps turn into human-readable form

louise-davies replied:
No problem, I just thought if we were converting from a `Date` to a unix timestamp and then back to a `Date`again it might be easier to just keep it as `Date`, but yeah it does complicate the types a little. The other thing was it meant I didn't have to match the timestamp column name to it's data in the select function (as you could infer timestamp from the type of `Date`), but you've fixed that :)

Commit:
DSEGOG-52 minor test improvements

louise-davies commented on src/api/records.tsx:
```diff
@@ -194,12 +195,12 @@ export const useRecordsPaginated = (): UseQueryResult<
   switch (axisName) {
     case 'timestamp':
-      formattedData = parseISO(record.metadata.timestamp);
+      formattedData = new Date(record.metadata.timestamp).getTime();
```
Is it not better to keep using the `parseISO` function as that's what `useRecordsPaginated` uses? Just for consistency

sam-glendenning replied:
Okay so I had to look this up to double check exactly what the difference is. As I understand it, `parseISO` takes into account ISO-formatted date-time strings, which is basically what the API is giving us right now. So `parseISO` is the best function here and I'll switch that out to make use of it.

In this instance, we specifically need this date-time value as a unix timestamp, hence why I also call `getTime()` on it. But I presume you're referring specifically to the use of `new Date()` and not this part.

louise-davies replied:
Yeah I was referring to the `new Date` bit 👍

louise-davies reviewed (changes requested):
Seems to be working fine - only problem is that when a user adds a channel, they have to clear the channel select box before they can type in a new channel. Could we auto clear this?

Commit:
DSEGOG-52 harmonising use of parseISO

Removing usage of new Date(timestamp) in tests using backend-retrieved data. This makes tests more reflective of backend data, which uses ISO-formatted timestamps

Commit:
DSEGOG-52 auto-clearing autocomplete and fixing channel removal bug

Autocompletes now clear when the user presses enter OR selects a valid option. Also fixing a bug where the user could repeatedly add the same channel to a plot, which caused confusion when trying to remove it from the plot

louise-davies reviewed (approved):
That's a lot better now that we don't have to clear the text box. Also, good catch on the duplicate channels!

Looks good :)
